From abdc9d99e44429124886d6ae1f783ad961cdb8d6 Mon Sep 17 00:00:00 2001
From: dineshsv <dineshsv@ami.com>
Date: Tue, 18 Jul 2023 13:34:11 +0530
Subject: [PATCH] SEL entries used percentage showing unknown

Tested:

-Verified SEL entries used percentage showing is working as expected.

```shell

Before fix - SEL entries used percentage showing unknown

root@intel-obmc:~# ipmitool sel info
SEL Information
Version : 1.5 (v1.5, v2 compliant)
Entries : 63
Free Space : 65535 bytes or more
Percent Used : unknown
Last Add Time : 06/22/23 06:35:09 UTC
Last Del Time : Not Available
Overflow : false
Supported Cmds : 'Reserve'
root@intel-obmc:~#

After fix - SEL entries used percentage showing, working as expected.

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.126 -U root -P 0penBmc -I lanplus -C 17 sel info
SEL Information
Version          : 1.5 (v1.5, v2 compliant)
Entries          : 200
Free Space       : 28800 bytes
Percent Used     : 10%
Last Add Time    : 07/18/2023 07:03:19
Last Del Time    : 07/18/2023 06:57:56
Overflow         : false
Supported Cmds   : 'Reserve'

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.126 -U root -P 0penBmc -I lanplus -C 17 sel clear
Clearing SEL.  Please allow a few seconds to erase.
msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.126 -U root -P 0penBmc -I lanplus -C 17 sel info
SEL Information
Version          : 1.5 (v1.5, v2 compliant)
Entries          : 0
Free Space       : 32000 bytes
Percent Used     : 0%
Last Add Time    : Not Available
Last Del Time    : 07/18/2023 07:05:10
Overflow         : false
Supported Cmds   : 'Reserve'

msgubuntu@Dell-server-3:~$ ipmitool -H 10.0.98.126 -U root -P 0penBmc -I lanplus -C 17 sel info
SEL Information
Version          : 1.5 (v1.5, v2 compliant)
Entries          : 2000
Free Space       : 0 bytes
Percent Used     : 100%
Last Add Time    : 07/18/2023 07:55:40
Last Del Time    : 07/18/2023 07:05:10
Overflow         : false
Supported Cmds   : 'Reserve'
msgubuntu@Dell-server-3:~$

```

Signed-off-by: dineshsv <dineshsv@ami.com>
---
 src/storagecommands.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/storagecommands.cpp b/src/storagecommands.cpp
index 1a110ee..18cd880 100644
--- a/src/storagecommands.cpp
+++ b/src/storagecommands.cpp
@@ -43,7 +43,7 @@ namespace intel_oem::ipmi::sel
 {
 static const std::filesystem::path selLogDir = "/var/log";
 static const std::string selLogFilename = "ipmi_sel";
-
+static const uint16_t maxSELEntries = 2000;
 static int getFileTimestamp(const std::filesystem::path& file)
 {
     struct stat st;
@@ -943,8 +943,9 @@ ipmi::RspType<uint8_t,  // SEL version
     uint32_t eraseTimeStamp = intel_oem::ipmi::sel::erase_time::get();
     constexpr uint8_t operationSupport =
         intel_oem::ipmi::sel::selOperationSupport;
-    constexpr uint16_t freeSpace =
-        0xffff; // Spec indicates that more than 64kB is free
+
+    uint16_t freeSpace = (intel_oem::ipmi::sel::maxSELEntries - entries) *
+                         ipmi::sel::selRecordSize;
 
     return ipmi::responseSuccess(selVersion, entries, freeSpace, addTimeStamp,
                                  eraseTimeStamp, operationSupport);
-- 
2.25.1

